state-chain/primitives/src/lib.rs

```diff
@@ -145,7 +131,7 @@ pub type Price = U256;
 	Copy,
 	Default,
 )]
-pub struct PriceLimits {
+pub struct PriceLimits<Price> {
```

Comment:
This made me think that we instantiate `PriceLimits` with different types for `Price`, but now I realise that this is only to avoid importing the actual type from `amm_math`? Maybe it is better to reorganise code somewhat, like moving `Price` itself into cf_primitives (seems appropriate?) or alternatively moving `PriceLimits` next to `Price`.